src/pymor/algorithms/timestepping.py

```diff
@@ -10,221 +10,602 @@
+                t_next = self._next_interpolation_point
+                U_next = self._interpolate(t_next)
+                self._next_interpolation_point += self._save_dt
+                if self.return_times:
+                    return U_next, t_next
+                else:
+                    return U_next
+        else:
+            # the trajectory is requested as is
+            U, self.t = self._step()
+            if self.return_times:
+                return U, t
+            else:
+                return U
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self._interpolated_step()
+
+
+class SingleStepTimeStepperIterator(TimeStepperIterator):
+    """Base class for iterators of single-step methods.
+
+    See :meth:`TimeStepperIterator` for a documentation of the init parameters.
+
+    Note: derived classes only have to implement :meth:`_step_function`, and optionally :meth:`_interpolate`
+    """
+
+    @abstractmethod
+    def _step_function(self, U_n, t_n):
+        pass
+
+    def _step(self):
+        if not hasattr(self, 'U_n'):
+            if not self.logging_disabled:
+                self.logger.debug(f't={self.t}: returning initial data (mu={self.mu}) ...')
+            self.t_n = self.t0
+            self.U_n = self.U0.copy()
+            return self.U_n, self.t_n
+        elif not hasattr(self, 'U_np1'):
+            # this is the first step
+            if not self.logging_disabled:
+                self.logger.debug(f't={self.t}: stepping (mu={self.mu}) ...')
+            self.U_np1, self.t_np1 = self._step_function(self.U_n, self.t_n)
+            return self.U_np1, self.t_np1
+        else:
+            # this is a usual step
+            self.t_n = self.t_np1
+            self.U_n = self.U_np1.copy()
+            if not self.logging_disabled:
+                self.logger.debug(f't={self.t}: stepping (mu={self.mu}) ...')
+            self.U_np1, self.t_np1 = self._step_function(self.U_n, self.t_n)
+            return self.U_np1, self.t_np1
+
+
+class ImplicitEulerIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, iter, return_times)
+        self.__auto_init(locals())
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the step function U_np1 = (M + dt A)^{-1}(M U_n + dt F)
+        M_dt_A = (M + A * dt).with_(
+                solver_options=A.solver_options if stepper.solver_options == 'operator' else \
+                               M.solver_options if stepper.solver_options == 'mass' else \
+                               stepper.solver_options)
+        if not M_dt_A.parametric or 't' not in M_dt_A.parameters:
```

Comment:
```suggestion
        if 't' not in M_dt_A.parameters:
```

Nowadays, `parameters` never is `None`.